client/my-sites/upgrades/map-domain/index.jsx

```diff
@@ -1,136 +1,146 @@
-			errorMessage: null
-		};
-	},
-
-	componentWillMount: function() {
-		this.checkSiteIsUpgradeable();
-	},
-
-	componentDidMount: function() {
-		if ( this.props.sites ) {
-			this.props.sites.on( 'change', this.checkSiteIsUpgradeable );
-		}
-	},
-
-	componentWillUnmount: function() {
-		if ( this.props.sites ) {
-			this.props.sites.off( 'change', this.checkSiteIsUpgradeable );
-		}
-	},
-
-	checkSiteIsUpgradeable: function( ) {
-		if ( ! this.props.sites ) {
-			return;
-		}
-
-		const selectedSite = this.props.sites.getSelectedSite();
-
-		if ( selectedSite && ! selectedSite.isUpgradeable() ) {
-			page.redirect( '/domains/add/mapping' );
-		}
-	},
-
-	goBack: function() {
-		const selectedSite = this.props.sites && this.props.sites.getSelectedSite();
 		if ( ! selectedSite ) {
-			page( this.props.path.replace( '/mapping', '' ) );
+			page( '/domains/add' );
 			return;
 		}
 
 		if ( selectedSite.is_vip ) {
-			page( paths.domainManagementList( selectedSite.slug ) );
+			page( paths.domainManagementList( selectedSiteSlug ) );
 			return;
 		}
 
-		page( '/domains/add/' + selectedSite.slug );
-	},
+		page( '/domains/add/' + selectedSiteSlug );
+	};
+
+	handleRegisterDomain = ( suggestion ) => {
+		const { selectedSiteSlug } = this.props;
 
-	handleRegisterDomain( suggestion ) {
 		upgradesActions.addItem(
 			cartItems.domainRegistration( {
 				productSlug: suggestion.product_slug,
 				domain: suggestion.domain_name
 			} )
 		);
 
-		if ( this.isMounted() ) {
-			page( '/checkout/' + this.props.sites.getSelectedSite().slug );
-		}
-	},
+		page( '/checkout/' + selectedSiteSlug );
+	};
 
-	handleMapDomain( domain ) {
-		const selectedSite = this.props.sites.getSelectedSite();
+	handleMapDomain = ( domain ) => {
+		const { selectedSite, selectedSiteSlug } = this.props;
 
 		this.setState( { errorMessage: null } );
 
 		// For VIP sites we handle domain mappings differently
 		// We don't go through the usual checkout process
 		// Instead, we add the mapping directly
 		if ( selectedSite.is_vip ) {
-			wpcom.addVipDomainMapping( selectedSite.ID, domain ).then( () => {
-				page( paths.domainManagementList( selectedSite.slug ) );
-			}, error => this.setState( { errorMessage: error.message } ) );
+			wpcom.addVipDomainMapping( selectedSite.ID, domain )
+				.then(
+					() => page( paths.domainManagementList( selectedSiteSlug ) ),
+					( error ) => this.setState( { errorMessage: error.message } )
+				);
 			return;
 		}
 
 		upgradesActions.addItem( cartItems.domainMapping( { domain } ) );
 
-		if ( this.isMounted() ) {
-			page( '/checkout/' + selectedSite.slug );
-		}
-	},
+		page( '/checkout/' + selectedSiteSlug );
+	};
 
-	render: function() {
-		let selectedSite;
+	componentWillMount() {
+		this.checkSiteIsUpgradeable( this.props );
+	}
 
-		if ( this.props.sites ) {
-			selectedSite = this.props.sites.getSelectedSite();
+	componentWillReceiveProps( nextProps ) {
+		if ( nextProps.selectedSiteId !== this.props.selectedSiteId ) {
```

Comment:
Good catch there, although it's a rare case, it's a perfectly valid scenario, so let's remove the check. I've suggested that in #13399.